app/javascript/src/PaymentGateways/braintree/BraintreeForm.jsx

```diff
@@ -57,14 +57,12 @@ const BraintreeForm = ({
   }, [braintreeNonceValue])
 
   const get3DSecureError = (response) => {
-    if (
-      response.name === 'BraintreeError' ||
-      response.threeDSecureInfo.status !== 'authenticate_successful' ||
-      response.threeDSecureInfo.status !== 'authenticate_attempt_successful'
-    ) {
-      return CC_ERROR_MESSAGE
-    }
-    return null
+    return (
+      !response.threeDSecureInfo ||
+      !!response.threeDSecureInfo.status.match(/authenticate_(attempt_)?successful/) === false
```

Comment:
So we are confident, without having types, that `response.threeDSecureInfo.status` won't be `undefined`?

Reply:
Unfortunately, there are no types available, but checking the docs and doing a lot of testing, I think yes.
When `response.threeDSecureInfo`, it will contain `status`

Reply:
And what does the documentation about their API say?

Reply:
Well, you need to check several docs, but you can start with
General reference: https://braintree.github.io/braintree-web/current/ThreeDSecure.html
Status codes: https://developer.paypal.com/braintree/docs/guides/3d-secure/server-side#status-codes
Testing: https://developer.paypal.com/braintree/docs/guides/3d-secure/testing-go-live

Reply:
I'd rather trust you have, indeed, read it and can assure us that not checking for `status` `undefined` is possible since their docs says so... instead of pausing my tasks to read their several docs. Also, if it was up to me, I'd just check if `undefined` and report an error.... :)

Reply:
+1 for checking `undefined` regardless of docs.

Reply:
Ok, checking if undefined, see line 62

Reply:
One last suggestion that looks cleaner IMHO:

```javascript
const get3DSecureError = (response) => {
  const { threeDSecureInfo: { status: message } = { status: null } } = response
  return message && message.match(/authenticate_(attempt_)?successful/) ? null : CC_ERROR_MESSAGE
}
```

Reply:
@guicassolato , this looks awesome, I'm impressed by Destructuring assignment syntax.

Reply:
you could go even cleaner and do:

```js
const get3DSecureError = ({ threeDSecureInfo: { status: message } = { status: null } } ) => (
  message && message.match(/authenticate_(attempt_)?successful/) ? null : CC_ERROR_MESSAGE
)
```

Reply:
@akostadinov JS could be elegant AF ;)

Reply:
Great suggestion!
I'll go with @guicassolato 's one, is more readable and will be easy to add types to the response in the future